### PR TITLE
More dependency automation and IDEA configuration

### PR DESCRIPTION
**Pull request by dmdashenkov**
In this PR we add some more automation features to the plugin.

### Library dependencies

Now, in order to add a Spine library dependency, the user can use API provided by the plugin, instead of specifying artifacts manually. This includes:
 - the Google Cloud Datastore storage implementation:
```groovy
spine.enableJava {
    withDatastore()
}
```
 - web API library:
```groovy
spine.enableJava().webServer()
```
 - web library with integration over the Firebase Realtime Database:
```groovy
spine.enableJava().firebaseWebServer()
```

### IDEA configuration

If both Bootstrap and the `idea` plugin are applied, Bootstrap configures the `idea` plugin, so that the IDE displays sources and generated sources as expected.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov, PTAL.

File: plugin/src/main/java/io/spine/tools/gradle/config/ArtifactSnapshot.java [hunk 1]
```diff
@@ -80,6 +84,8 @@ private static ArtifactSnapshot load() {
                 .setSpineCoreVersion(properties.getProperty("spine.version.core"))
```
Comment (alexander-yevsyukov) [inline plugin/src/main/java/io/spine/tools/gradle/config/ArtifactSnapshot.java:92]: Please see if we can have a class which declares these constants and used at both ends.
If it's not easy, please document usage of this constants at both ends: when they generated, and here (consumed), referring to the other end of this trick.

**Review by alexander-yevsyukov — CHANGES_REQUESTED** [review]
Please see my comment.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov, PTAL again.

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM

**Pull request merged**